src/rules/avoid-importing-barrel-files.ts

```diff
@@ -0,0 +1,332 @@
+import { readFileSync } from 'node:fs'
+import { builtinModules } from 'node:module'
+import path from 'node:path'
+
+import {
+  count_module_graph_size,
+  is_barrel_file,
+} from 'eslint-barrel-file-utils/index.cjs'
```

Comment:
Hmm, why `.cjs` is referenced specifically? Without `/index.cjs` it won't work? Then that's an issue of `eslint-barrel-file-utils`? I'm wondering should `eslint-barrel-file-utils` be merged into `unrs-resolver` at the same time.

Reply:
from the OP:

> eslint-barrel-file-utils is missing a d.cts for index.cjs, and we can't import index.js (or bare) because the index.d.ts exported names mismatch the ones in index.d.ts

this is why. maybe lets see what @thepassle thinks too

Reply:
I'm wondering should it be merged into `unrs-resolver` instead, at least as a crate dep.